test/insertion_order_parser_test.dart

```diff
@@ -0,0 +1,293 @@
+import 'package:angular_test/angular_test.dart';
+import 'package:test/test.dart';
+
+import 'package:dv360_excel_plugin/src/insertion_order_parser.dart';
+import 'package:dv360_excel_plugin/src/proto/insertion_order.pb.dart';
+
+void main() {
+  group(InsertionOrderParser, () {
+    String input;
+    const emptyEntry = '';
+
+    InsertionOrder insertionOrderTemplate;
+    InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange
+        oneDateRange;
+    InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date
+        oneStartDate;
+    InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date
+        oneEndDate;
+
+    setUp(() {
+      insertionOrderTemplate = InsertionOrder()
+        ..advertiserId = emptyEntry
+        ..campaignId = emptyEntry
+        ..insertionOrderId = emptyEntry
+        ..displayName = emptyEntry
+        ..updateTime = emptyEntry
+        ..entityStatus = InsertionOrder_EntityStatus.ENTITY_STATUS_UNSPECIFIED
+        ..pacing = InsertionOrder_Pacing()
+        ..budget = InsertionOrder_InsertionOrderBudget();
+
+      oneStartDate =
+          InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date()
+            ..year = 2019
+            ..month = 1
+            ..day = 1;
+
+      oneEndDate =
+          InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange_Date()
+            ..year = 2019
+            ..month = 12
+            ..day = 31;
+
+      oneDateRange =
+          InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment_DateRange()
+            ..startDate = oneStartDate
+            ..endDate = oneEndDate;
+    });
+
+    tearDown(disposeAnyRunningTest);
+
+    group('parse insertionOrder from empty json:', () {
+      setUp(() => input = '{}');
+
+      test('result is an empty insertionOrder instance', () {
+        final actual = InsertionOrderParser.parse(input);
+
+        final expected = InsertionOrder();
+        expect(actual, expected);
+      });
+    });
+
+    group('parse insertionOrder from json that contains only advertiserId:',
+        () {
+      setUp(() => input = '{"advertiserId":"111111"}');
+
+      test('result is an insertionOrder with only advertiserId', () {
+        final actual = InsertionOrderParser.parse(input);
+
+        final expected = insertionOrderTemplate..advertiserId = '111111';
+        expect(actual, expected);
+      });
+    });
+
+    group('parse insertionOrder from json that contains everything:', () {
```

Comment:
group are used to share setup/teardown between multiple tests and logically group them.
if you have groups of 1 test, consider putting in all of the logic inside the 1 test in a traditional arrange,act, assert setup.

e.g. so if you had another test that verified something about the complete input then it would make sense to share the setup